### PR TITLE
Add docs about cloud events structure

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

Add headers the payload format for cloud events generated by Tekton

Signed-off-by: Andrea Frittoli <andrea.frittoli@uk.ibm.com>

/kind documentation

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Review by dibyom — APPROVED** [review]

**Comment by bigkevmcd** [comment]
Thanks @afrittoli :-)

File: docs/events.md [hunk 1]
```diff
@@ -68,3 +68,129 @@ Resource      |Event    |Event Type
 `PipelineRun` | `Condition Change while Running` | `dev.tekton.event.pipelinerun.unknown.v1`
 `PipelineRun` | `Succeed` | `dev.tekton.event.pipelinerun.successful.v1`
 `PipelineRun` | `Failed`  | `dev.tekton.event.pipelinerun.failed.v1`
+
+## Format of `CloudEvents`
+
+According to the `CloudEvents` spec, HTTP headers are included to match the context fields. For example:
```
Comment (chmouel) [inline docs/events.md:74]: Maybe add a link to the spec? https://github.com/cloudevents/spec/blob/master/spec.md

**Comment by chmouel** [comment]
Thanks @afrittoli tiny comment about adding a link to the actual cloudevents spec on https://github.com/cloudevents/spec/blob/master/spec.md

File: docs/events.md (hunk 1 above)
Comment (afrittoli) [reply docs/events.md:74]: Thanks! Done

**Pull request merged**